src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/hotspot/HotSpotConstantPool.java

```diff
@@ -658,17 +663,18 @@ public Object lookupConstant(int cpi) {
                  * "pseudo strings" (arbitrary live objects) patched into a String entry. Such
                  * entries do not have a symbol in the constant pool slot.
                  */
-                return compilerToVM().resolvePossiblyCachedConstantInPool(this, cpi);
+                return compilerToVM().lookupConstantInPool(this, cpi, true);
```

Comment:
This is true because it's always safe to do so?

Reply:
I don't really know what these "pseudo strings" are. In any case, I don't think resolving them involves calling any Java code.

Reply:
Sounds good.  It's not currently causing any problems and we can investigate further if we start restricting when we `can_call_java` and problems show up here.